streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -382,7 +383,13 @@ public void setGlobalStateRestoreListener(final StateRestoreListener globalState
      * @return Map of all metrics.
      */
     public Map<MetricName, ? extends Metric> metrics() {
-        return Collections.unmodifiableMap(metrics.metrics());
+        final Map<MetricName, Metric> result = new LinkedHashMap<>();
+        for (final StreamThread thread : threads) {
+            result.putAll(thread.consumerMetrics());
+        }
+        if (globalStreamThread != null) result.putAll(globalStreamThread.consumerMetrics());
+        result.putAll(metrics.metrics());
+        return Collections.unmodifiableMap(result);
     }
```

Comment:
Note that I'm adding all the internal consumer metrics to the map returned by `metrics`. Is this ok?!?!?

I think so, since it's supposed to be a "map of all metrics", so I'd argue it's not a semantic change, and all the Streams metrics will still be there, so nothing will break.

But it's clearly not what we had before, so I want to ask explicitly.

I guess the alternative for anyone, like me, who wants to get the lag metrics is to register a metric reporter for the consumers and capture the metrics that way, but I would question the existence of this method if it only returns some of the metrics and you have to implement your own reporter to capture the others...

Reply:
I'm not sure seems fine to me, but I'll let others weigh in.

Reply:
Thanks for the catch.

I agree that we should be able to return all the embedded client's metrics as well; thinking about this a bit more, we should also include the producer metrics as well.

I'll go ahead and merge this PR as is while leaving a TODO marker for @vvcephei